### ISSUE TITLE
System freezes until force reset reason unknown

### ISSUE THREAD

**Issue by GDBlaster** [pop-os/pop#3130]
**Distribution (run `cat /etc/os-release`):**
 Pop!_OS 22.04 LTS x86_64
 
 
**Related Application and/or Package Version (run `apt policy $PACKAGE NAME`):**

**Issue/Bug Description:**

after 30mins to an hour of use the system freezes completely until manually reset since latest kernel update

**Steps to reproduce (if you know):**

it seems to happen randomly

**Expected behavior:**

not freezing

**Other Notes:**

music continues playing even during the freeze

specs:
- Acer Aspire A515-57 V1.15
- i5 1235U 12core
- 8GB RAM
- IGPU

**Comment by rossanodr** [comment]
Same here

**Comment by fcamp** [comment]
Same here.

Some minutes ago it freezed during a slack call. Audio went on working fine, but only mouse pointer continued to move, even if I couldn't click anything. 
I previously created a keyboard shortcut in order to reset gnome `killall -3 gnome-shell`, but it doesn't work when the pc freezes.

**Comment by leviport** [comment]
This sounds like it could be the same as a bug I've been investigating lately:  https://gitlab.freedesktop.org/drm/intel/-/issues/8421

If it is the same, the good news is that a much simpler workaround can unfreeze your session without losing any progress. Switching to TTY3 (ctrl + alt + F3) and back to TTY2 (ctrl + alt + F2) usually gets me back to an unfrozen session after a bit. It usually takes about 15 - 20 seconds on my machine, and I have to hit ctrl + alt + F3 quite a few times, but at least I don't have to start a new session and re-open all of my applications.

**Comment by GDBlaster** [comment]
> This sounds like it could be the same as a bug I've been investigating lately: https://gitlab.freedesktop.org/drm/intel/-/issues/8421
> 
> If it is the same, the good news is that a much simpler workaround can unfreeze your session without losing any progress. Switching to TTY3 (ctrl + alt + F3) and back to TTY2 (ctrl + alt + F2) usually gets me back to an unfrozen session after a bit. It usually takes about 15 - 20 seconds on my machine, and I have to hit ctrl + alt + F3 quite a few times, but at least I don't have to start a new session and re-open all of my applications.

shortcuts to switch TTY are unresponsive too

**Comment by leviport** [comment]
It could be a different issue then, but like I said, it takes a while for my machine to respond. After the first TTY hotkey hit, I have to wait about 20 seconds, then hit it a few more times. It's slow, but at least it doesn't destroy the session.

**Comment by leviport** [comment]
Also if you look in your journal logs and see a message along the lines of `flip_done() timed out`, then I'm pretty sure it's the same bug.

**Comment by leviport** [comment]
Switching TTYs causes a reboot? This sounds quite different from what I'm seeing.

**Comment by fcamp** [comment]
Hmmm, quite strange. 
Yesterday I booted with the current kernel version, I was not at home, but the only things that changed were the displays I attached the laptop to. I've had no issues.

Today I'm at the office as well, booted with the current kernel. I expect no issues, let's see

**Comment by fcamp** [comment]
@leviport Yes, I can confirm that I found many `flip_done() timed out` occurrences in the logs.

```
Sep 15 08:53:43 pop-os kernel: i915 0000:00:02.0: [drm] *ERROR* flip_done timed out
Sep 15 08:53:43 pop-os kernel: i915 0000:00:02.0: [drm] *ERROR* [CRTC:131:pipe B] commit wait timed out
Sep 15 08:53:53 pop-os kernel: i915 0000:00:02.0: [drm] *ERROR* flip_done timed out
Sep 15 08:53:53 pop-os kernel: i915 0000:00:02.0: [drm] *ERROR* [CONNECTOR:245:HDMI-A-1] commit wait timed out
Sep 15 08:54:03 pop-os kernel: i915 0000:00:02.0: [drm] *ERROR* flip_done timed out
Sep 15 08:54:03 pop-os kernel: i915 0000:00:02.0: [drm] *ERROR* [PLANE:82:plane 1B] commit wait timed out
Sep 15 08:54:13 pop-os kernel: i915 0000:00:02.0: [drm] *ERROR* [CRTC:131:pipe B] flip_done timed out
Sep 15 08:54:24 pop-os kernel: i915 0000:00:02.0: [drm] *ERROR* flip_done timed out
Sep 15 08:54:24 pop-os kernel: i915 0000:00:02.0: [drm] *ERROR* [CRTC:131:pipe B] commit wait timed out
Sep 15 08:54:34 pop-os kernel: i915 0000:00:02.0: [drm] *ERROR* flip_done timed out
Sep 15 08:54:34 pop-os kernel: i915 0000:00:02.0: [drm] *ERROR* [CONNECTOR:245:HDMI-A-1] commit wait timed out
Sep 15 08:54:44 pop-os kernel: i915 0000:00:02.0: [drm] *ERROR* flip_done timed out
Sep 15 08:54:44 pop-os kernel: i915 0000:00:02.0: [drm] *ERROR* [PLANE:82:plane 1B] commit wait timed out
Sep 15 08:54:54 pop-os kernel: i915 0000:00:02.0: [drm] *ERROR* [CRTC:131:pipe B] flip_done timed out
```
However it's quite strange: it happens *only* when I connect my old hdmi display at home using the current kernel version.
Now I'm using kernel `Linux 6.2.6-76060206-generic x86_64` and I had no issues

**Comment by AMoonRabbit** [comment]
I am seeing this again on latest kernels. This is only on shutdown, with a USB-C to HDMI adaptor (Framework hardware). 
```
i915 0000:00:02.0: [drm] *ERROR* [CRTC:80:pipe A] flip_done timed out
i915 0000:00:02.0: [drm] *ERROR* [CRTC:131:pipe B] flip_done timed out
i915 0000:00:02.0: [drm] *ERROR* flip_done timed out
i915 0000:00:02.0: [drm] *ERROR* [CRTC:80:pipe A] commit wait timed out
i915 0000:00:02.0: [drm] *ERROR* [CRTC:131:pipe B] commit wait timed out
i915 0000:00:02.0: [drm] *ERROR* [CONNECTOR:236:eDP-1] commit wait timed out
i915 0000:00:02.0: [drm] *ERROR* [CONNECTOR:258:DP-2] commit wait timed out
i915 0000:00:02.0: [drm] *ERROR* [PLANE:31:plane 1A] commit wait timed out
i915 0000:00:02.0: [drm] *ERROR* [PLANE:82:plane 1B] commit wait timed out
i915 0000:00:02.0: [drm] *ERROR* [ENCODER:235:DDI A/PHY A] PPS 0 panel status timeout: PP_STATUS: 0x80000008 PP_CONTROL: 0x80000867
```

**Comment by diasbalmash** [comment]
@AMoonRabbit Hi. Did you solve your problem?

**Comment by AMoonRabbit** [comment]
@diasbalmash Not yet, I do have something to try though which is to force another i915 driver to load (adlp_dmc_ver2_16.bin) as suggested in this thread https://gitlab.freedesktop.org/drm/intel/-/issues/8421

I've added it with `kernelstub -a i915.dmc_firmware_path=/lib/firmware/i915/adlp_dmc_ver2_16.bin` but I have no idea if this has worked yet as I have not been in a position to test. Should be able to over the weekend.

**Comment by AMoonRabbit** [comment]
Huzzah! I think I have a work around. Slight change to the kernelstub entry as I misunderstood it's use. By setting `kernelstub -a i915.dmc_firmware_path=""` the hang on reboot/shutdown appears to no longer happen. So there is an issue with the files it usually loads.

**Comment by ljnsn** [comment]
I have the same issue, it has nothing to do with external monitors as mentioned in the i915 issue though, it happens without any monitor connected. Switching TTY works.
